plugins/inputs/systemd_units/systemd_units_linux.go

```diff
@@ -0,0 +1,227 @@
+package systemd_units
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// SystemdUnits is a telegraf plugin to gather systemd unit status
+type SystemdUnits struct {
+	Timeout   internal.Duration
+	UnitType  string `toml:"unittype"`
+	systemctl systemctl
+}
+
+type systemctl func(Timeout internal.Duration, UnitType string) (*bytes.Buffer, error)
+
+const measurement = "systemd_units"
+
+var (
+	defaultTimeout  = internal.Duration{Duration: time.Second}
+	defaultUnitType = "service"
+)
+
+// Description returns a short description of the plugin
+func (systemd_units *SystemdUnits) Description() string {
+	return "Gather systemd units state"
+}
+
+// SampleConfig returns sample configuration options.
+func (systemd_units *SystemdUnits) SampleConfig() string {
+	return `
+  ## The default timeout of 1s for systemctl execution can be overridden here:
+  # timeout = "1s"
+  # unittype = "service"
+`
+}
+
+// Gather parses systemctl outputs and adds counters to the Accumulator
+func (systemd_units *SystemdUnits) Gather(acc telegraf.Accumulator) error {
+	out, err := systemd_units.systemctl(systemd_units.Timeout, systemd_units.UnitType)
+	if err != nil {
+		return err
+	}
+
+	scanner := bufio.NewScanner(out)
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		data := strings.Fields(line)
+		if len(data) < 4 {
+			acc.AddError(fmt.Errorf("Error parsing line (expected at least 4 fields): %s", line))
+			continue
+		}
+		tags := map[string]string{
+			"name": data[0],
+		}
+		// unit_load_state_table, see
```

Comment:
Make these state table consts and pull them out of gather.

Reply:
I've moved them, but maps cannot be const!?